internal/resolver/dns/fake_net_resolver_test.go

```diff
@@ -47,9 +49,28 @@ func (tr *testNetResolver) LookupHost(ctx context.Context, host string) ([]strin
 	tr.mu.Lock()
 	defer tr.mu.Unlock()
 
+	// trigger a timeout if the host name contains "infinity"
+	if strings.Contains(host, "infinity") {
+		<-time.After(88 * time.Millisecond)
+	}
```

Comment:
Timing based tests like this are problematic, because if the machine hiccups for >88 ms, you could end up with a race where this case wins and "infinity" actually resolves.

Please instead use a `testNetResolver.lookupHostCh` that is unbuffered, and don't read from it until after the timeout occurs.